### DISCUSSION TITLE
dx serve fullstack desktop application

### DISCUSSION THREAD

**Discussion by luveti** [DioxusLabs/dioxus#2430] [Q&A] [answered]
Is there a way to do the following? `dx serve --platform fullstack --client-feature desktop`. Using `web` for `--client-feature` works, but I'm trying to create a desktop app that communicates with a server. I would also like the app to be usable from the web, but with less features than the desktop app.

The [Fullstack docs](https://dioxuslabs.com/learn/0.5/reference/fullstack/server_functions) use `cargo build` for things, but then one loses hot-reload.

**Comment by ealmloff** [comment, marked as answer]
You currently need to use cargo or two separate desktop builds when targeting fullstack + desktop. The fullstack platform in the CLI assumes that your client build targets wasm

It would be great to support fullstack+desktop builds in the CLI in the future
